python/cudf/cudf/tests/test_reshape.py

```diff
@@ -621,3 +621,32 @@ def test_crosstab_simple():
     expected = pd.crosstab(a, [b, c], rownames=["a"], colnames=["b", "c"])
     actual = cudf.crosstab(a, [b, c], rownames=["a"], colnames=["b", "c"])
     assert_eq(expected, actual, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "values", ["z", "z123", ["z123"], ["z", "z123", "123z"]]
+)
+def test_pivot_values(values):
+    data = [
+        ["A", "a", 0, 0, 0],
+        ["A", "b", 1, 1, 1],
+        ["A", "c", 2, 2, 2],
+        ["B", "a", 0, 0, 0],
+        ["B", "b", 1, 1, 1],
+        ["B", "c", 2, 2, 2],
+        ["C", "a", 0, 0, 0],
+        ["C", "b", 1, 1, 1],
+        ["C", "c", 2, 2, 2],
+    ]
+    columns = ["x", "y", "z", "z123", "123z"]
+    pdf = pd.DataFrame(data, columns=columns)
+    cdf = cudf.DataFrame(data, columns=columns)
+    expected = pd.pivot(pdf, index="x", columns="y", values=values)
+    actual = cudf.pivot(cdf, index="x", columns="y", values=values)
+    if not isinstance(values, (list, tuple)):
+        actual.columns = actual.columns.droplevel(0)
```

Comment:
Why is this needed? This looks like an incompatibility between pandas and cudf that we should resolve, but I don't know what this line does.

Reply:
I noticed that presently even when `values` is not a list but just a single character string(which was supported even before this change) we were returning a multiindex dataframe unlike Pandas. Assuming this is the expected behavior as it already existed, I added this in the test.

For example, if `values='z'`
This returns a single index DataFrame in Pandas but a MultiIndex DataFrame in cuDF

But if `values=[z]`
This returns a MultiIndex DataFrame in both Pandas and cuDF.

Reply:
Yup! This is actually part of the same bug you're trying to solve, I think. Strings are iterables of characters, but we want to handle strings as standalone objects, to match pandas.

You might need something like:
```python
if not isinstance(values, (list, tuple, str)):
    values = [values]
```
to prevent `'z'` from being converted to `['z']` and becoming a MultiIndex.

Reply:
I've made a similar change for the multicharacter string issue but I don't think that affects the MultiIndex issue as the dataframe gets created in a different helper function called `_pivot` which is also used in `unstack`. I'll take a look if that affects the expected behavior in `unstack` as well.

Reply:
I've made the change in `pivot` itself as unlike it for `unstack`, Pandas returns a MultiIndex dataframe itself even for string inputs.